x-pack/legacy/plugins/license_management/public/register_route.ts

```diff
@@ -87,6 +87,7 @@ if (licenseManagementUiEnabled) {
                   const unmountApp = await app.mount({ ...npStart } as any, {
                     element,
                     appBasePath: '',
+                    onAppLeave: () => undefined,
```

Comment:
It seems like in many places, an empty function is used.
Why not make `onAppLeave` optional?

Reply:
It's actually always provided by core `application` service for NP apps and is not meant to be optional. I was forced to add a noop function in legacy shims that manually mimic the core application service API to mount their NP app from legacy. Fortunately this will go away once everything is migrated.